core/trino-main/src/main/java/io/trino/sql/gen/columnar/FilterReorderingProfiler.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.gen.columnar;
+
+import it.unimi.dsi.fastutil.ints.IntArrays;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public final class FilterReorderingProfiler
+{
+    private static final long MIN_SAMPLE_POSITIONS_REORDERING = 4095;
```

Comment:
is 4095 meaningful?

Reply:
Orc and parquet readers produce pages in batches of 1, 2, 4, 8, 16,..., 8192.
4095 is 12 such initial pages

Reply:
Maybe add a comment. 
You may also define it as 1 + 1<<2 + 1<<3 + ... + 1<<11;